staking/programs/staking/src/context.rs

```diff
@@ -278,6 +279,97 @@ pub struct CreateTarget<'info> {
     pub system_program:    Program<'info, System>,
 }
 
+#[derive(Accounts)]
+#[instruction(amount : u64, recipient : Pubkey)]
+pub struct RequestSplit<'info> {
+    // Native payer:
+    #[account(mut, address = stake_account_metadata.owner)]
+    pub payer:                       Signer<'info>,
+    // Stake program accounts:
+    pub stake_account_positions:     AccountLoader<'info, positions::PositionData>,
+    #[account(seeds = [STAKE_ACCOUNT_METADATA_SEED.as_bytes(), stake_account_positions.key().as_ref()], bump = stake_account_metadata.metadata_bump)]
+    pub stake_account_metadata:      Account<'info, stake_account::StakeAccountMetadataV2>,
+    #[account(init_if_needed, payer = payer, space=split_request::SplitRequest::LEN ,  seeds = [SPLIT_REQUEST.as_bytes(), stake_account_positions.key().as_ref()], bump)]
+    pub stake_account_split_request: Account<'info, split_request::SplitRequest>,
+    #[account(seeds = [CONFIG_SEED.as_bytes()], bump = config.bump)]
+    pub config:                      Account<'info, global_config::GlobalConfig>,
+    // Primitive accounts :
+    pub system_program:              Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct AcceptSplit<'info> {
+    // Native payer:
+    #[account(mut, address = config.pda_authority)]
+    pub payer:                               Signer<'info>,
+    // Current stake accounts:
+    #[account(mut)]
+    pub current_stake_account_positions:     AccountLoader<'info, positions::PositionData>,
+    #[account(mut, seeds = [STAKE_ACCOUNT_METADATA_SEED.as_bytes(), current_stake_account_positions.key().as_ref()], bump = current_stake_account_metadata.metadata_bump)]
+    pub current_stake_account_metadata: Box<Account<'info, stake_account::StakeAccountMetadataV2>>,
+    #[account(seeds = [SPLIT_REQUEST.as_bytes(), current_stake_account_positions.key().as_ref()], bump)]
+    pub current_stake_account_split_request: Box<Account<'info, split_request::SplitRequest>>,
+    #[account(
+        mut,
+        seeds = [CUSTODY_SEED.as_bytes(), current_stake_account_positions.key().as_ref()],
+        bump = current_stake_account_metadata.custody_bump,
+    )]
+    pub current_stake_account_custody:       Account<'info, TokenAccount>,
+    /// CHECK : This AccountInfo is safe because it's a checked PDA
+    #[account(seeds = [AUTHORITY_SEED.as_bytes(), current_stake_account_positions.key().as_ref()], bump = current_stake_account_metadata.authority_bump)]
+    pub current_custody_authority:           AccountInfo<'info>,
+
+    // New stake accounts :
```

Comment:
is the intention here that these accounts will already exist before this instruction is called?

if so, consider a naming convention like "src_" and "dst_". The current names make it sound like the "new_" accounts will be created.

Reply:
(I also would appreciate some docs on the instruction methods below explaining how this process works, as I clearly don't understand it)

Reply:
Changed name to source and added some instructions